velox/functions/prestosql/types/TimestampWithTimeZoneType.h

```diff
@@ -145,4 +145,12 @@ inline int64_t pack(int64_t millisUtc, int16_t timeZoneKey) {
 inline Timestamp unpackTimestampUtc(int64_t dateTimeWithTimeZone) {
   return Timestamp::fromMillis(unpackMillisUtc(dateTimeWithTimeZone));
 }
+
+inline std::tuple<StringView, std::string_view>
+splitTimestampWithTimezoneString(const std::string_view& tsTzStr) {
```

Comment:
You cannot return a StringView from local string variables. The local variables will be destroyed on function return and the StrinView values will be invalid.
Let's remove this function and inline the implementation inside `castFromString`.

Reply:
Alright, inlined in the cast function.

Reply:
Replaced with a call to a similar, more comprehensive function Pedro recently added in `TimestampConversion.cpp` - `fromTimestampWithTimezoneString()` parses the tstz string and returns a pair of integers, no worries about lost references.